suite-native/transactions/src/components/TransactionDetail/TransactionDetailData.tsx

```diff
@@ -1,27 +1,41 @@
 import React from 'react';
 import { useSelector } from 'react-redux';
 
-import { Box, Card, Divider, Text, VStack } from '@suite-native/atoms';
-import { AccountKey, WalletAccountTransaction } from '@suite-common/wallet-types';
+import { Box, Card, Text, VStack } from '@suite-native/atoms';
+import { AccountKey } from '@suite-common/wallet-types';
 import { Icon } from '@trezor/icons';
 import { useFormatters } from '@suite-common/formatters';
 import { CryptoAmountFormatter, CryptoToFiatAmountFormatter } from '@suite-native/formatters';
 import { selectTransactionBlockTimeById, TransactionsRootState } from '@suite-common/wallet-core';
+import { EthereumTokenTransfer, WalletAccountTransaction } from '@suite-native/ethereum-tokens';
 
 import { TransactionDetailSummary } from './TransactionDetailSummary';
 import { TransactionDetailRow } from './TransactionDetailRow';
+import { TransactionDetailIncludedCoins } from './TransactionDetailIncludedCoins';
 
 type TransactionDetailDataProps = {
     transaction: WalletAccountTransaction;
     accountKey: AccountKey;
+    tokenTransfer?: EthereumTokenTransfer;
 };
 
-export const TransactionDetailData = ({ transaction, accountKey }: TransactionDetailDataProps) => {
+export const TransactionDetailData = ({
+    transaction,
+    accountKey,
+    tokenTransfer,
+}: TransactionDetailDataProps) => {
     const { DateTimeFormatter } = useFormatters();
     const transactionBlockTime = useSelector((state: TransactionsRootState) =>
         selectTransactionBlockTimeById(state, transaction.txid, accountKey),
     );
 
+    const isTokenTransactionDetail = !!tokenTransfer;
+    const transactionTokensCount = transaction.tokens?.length || 0;
+
+    const includedCoinsDisplayed =
+        (isTokenTransactionDetail && transactionTokensCount - 1 > 0) ||
```

Comment:
This conditions seems uncessary can you explain and split them into more variables with some names that will explain what they are doing?

Reply:
refactored to be clearer in 1e27599ac